src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -51,9 +50,12 @@ public class FunctionMetadataManager : IFunctionMetadataManager
 
             // Every time script host is re-intializing, we also need to re-initialize
             // services that change with the scope of the script host.
-            scriptHostManager.HostInitializing += (s, e) =>
+            scriptHostManager.ActiveHostChanged += (s, e) =>
```

Comment:
For future reference -- I changed this due to a race during specialization while I was testing this. The point here is to refresh services on the new host, and ActiveHostChanged is much more appropriate for this and removes the race. The ActiveHostChanged API was added after this code, so it wasn't an option at the time.

Also, we have another component that does something similar, which makes me more comfortable (https://github.com/Azure/azure-functions-host/blob/8ceb05a89a4337f07264d4991545538a3e8b58a0/src/WebJobs.Script/Config/ActiveHostConfigurationSource.cs#L32)